src/main/scala/piconot/internal/ourDSL.scala

```diff
@@ -0,0 +1,132 @@
+package piconot.internal
+
+import java.io.File
+import scalafx.application.JFXApp
+import scala.language.postfixOps
+import picolib.maze.Maze
+import picolib.semantics._
+
+/**
+ * Notes on combining blocked + open (from tues night):
+ * can combine with "+" in the form "this.+(open(..), blocked(..))" without making our own wrapper class (simply unpack Surroundings and create new one),
+ * However, to get proper binary operator (with the form, "open(..) + blocked(..)", we need to create our own class that mimics Surrounding because we'd need to define the operator inside it.
+ * (^Unless I'm forgetting about something obvious)
+ * Now our tuples might contain Surroundings or OurSurroundings (if users only pass in an 'open', for example, the '+' method will never be called, so the OurSurroundings will never get converted to a Surroundings).
+ * So, we must cast both types of tuples to ShortRule.
+ */
+
+class ourDSL extends JFXApp {
+
+  // Initializes the picobot and runs graphics display
+  def picobot(mazeName: String)(rules: Seq[Rule]*) = {
+    val rulesList = rules.flatten.toList
+    val maze = Maze("resources" + File.separator + mazeName + ".txt")
+    object EmptyBot extends Picobot(maze,rulesList)
+      with TextDisplay with GUIDisplay
+    stage = EmptyBot.mainStage
+    EmptyBot.run()
+  }
+
+  // Allows users to group by state a series of rules
+  def state(stateName: State)(moves: ShortRule*): Seq[Rule] = {
+    moves.map(r => Rule(stateName, r.s, r.m, r.state))
+  }
+
+  // Natural language method to set move direction
+  def move(dir: MoveDirection): MoveDirection = dir
+  
+  // Natural language method to set end state
+  def to(end: String): State = end
+  
+  // Passes variatic MoveDirection and uses pattern matching to check using relative description
+  // which direction is blocked
+  def blocked(walls: MoveDirection*): OurSurroundings = {
+    var northWall, eastWall, westWall, southWall: RelativeDescription = Anything
+    walls.foreach { dir =>
+      dir match {
+        case North => northWall = Blocked
+        case East  => eastWall = Blocked
+        case West  => westWall = Blocked
+        case South => southWall = Blocked
+      }
+    }
+    OurSurroundings(northWall, eastWall, westWall, southWall)
+  }
+
+  // Similar to blocked, but checks if the relative description is open
+  def open(walls: MoveDirection*): OurSurroundings = {
+    var northWall, eastWall, westWall, southWall: RelativeDescription = Anything
```

Comment:
While probably not the worst thing in the world, you definitely could have done this without using _var_. Avoiding this would probably make the code more typical Scala code.